### PR TITLE
GH-38954: [Dev] Fix spelling

### LINKED ISSUE #38954: [Dev] Spelling errors identified by check-spelling
### Describe the bug, including details regarding any error messages, version, and platform.

The [check-spelling action](https://github.com/marketplace/actions/check-spelling) enabled me to identify some misspellings...

The misspellings have been reported at https://github.com/jsoref/apache-arrow/actions/runs/7008141484/attempts/1#summary-19063787889

### Component(s)

Developer Tools

### PR DESCRIPTION
**Pull request by jsoref**
### What changes are included in this PR?

Spelling fixes to dev/

### Are there any user-facing changes?

* Closes: #38954

### CODE REVIEW

File: dev/archery/archery/release/core.py [hunk 1]
```diff
@@ -468,7 +468,7 @@ def curate(self, minimal=False):
                 parquet.append((self.jira_instance.issue(c.issue), c))
             else:
                 warnings.warn(
-                    f'Issue {c.issue} is not MINOR nor pertains to GH' +
+                    f'Issue {c.issue} does not pertain to GH' +
```
Comment (kou) [inline dev/archery/archery/release/core.py:471]: Why do we need this?

**Review by kou — APPROVED** [review]
+1

File: dev/archery/archery/release/core.py (hunk 1 above)
Comment (jsoref) [reply dev/archery/archery/release/core.py:471]: The logic didn't match the message...
This is in an else chain for `if c.issue is None:` -- it doesn't know `if c.title.minor:`

**Review by pitrou — APPROVED** [review]

File: dev/archery/archery/release/core.py (hunk 1 above)
Comment (kou) [reply dev/archery/archery/release/core.py:471]: I see.

**Pull request merged**